overlord/snapstate/snapmgr.go

```diff
@@ -386,6 +418,96 @@ func (m *SnapManager) blockedTask(cand *state.Task, running []*state.Task) bool
 	return false
 }
 
+var CanAutoRefresh func(st *state.State) bool
+
+// ensureRefreshes ensures that we refresh all installed snaps periodically
+func (m *SnapManager) ensureRefreshes() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	// see if it even makes sense to try to refresh
+	if CanAutoRefresh == nil || !CanAutoRefresh(m.state) {
+		return nil
+	}
+
+	tr := config.NewTransaction(m.state)
+
+	// for tests
+	var refreshDisabled bool
+	err := tr.Get("core", "refresh.disabled", &refreshDisabled)
+	if err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	if refreshDisabled {
+		return nil
+	}
+
+	var lastRefresh time.Time
+	err = tr.Get("core", "refresh.last", &lastRefresh)
+	if err != nil && !config.IsNoOption(err) {
+		return err
+	}
+
+	var refreshRand time.Duration
+	err = tr.Get("core", "refresh.rand", &refreshRand)
```

Comment:
Can we please drop this?  Given that the syntax in the FIXME above supports windows, we can keep this as an internal var or const while we don't finish the implementation of the full scheduling.